extensions/blocks/mailchimp/view.js

```diff
@@ -41,22 +45,41 @@ function fetchSubscription( blogId, email, params ) {
 	} );
 }
 
+function validateEmail( form, emailField ) {
+	const email = emailField.value;
+	const errorClass = 'error';
+	emailField.classList.remove( errorClass );
+	if ( ! emailValidator.validate( email ) ) {
+		emailField.classList.add( errorClass );
+		if ( typeof document.createElement( 'input' ).reportValidity === 'function' ) {
+			// In case that browser supports it, trigger HTML5 validation
+			form.reportValidity();
+		}
+		return false;
+	}
+
+	return true;
+}
+
 function activateSubscription( block, blogId ) {
 	const form = block.querySelector( 'form' );
-	const errorClass = 'error';
+	const emailField = block.querySelector( 'input[name=email]' );
 	const processingEl = block.querySelector( '.' + blockClassName + '_processing' );
 	const errorEl = block.querySelector( '.' + blockClassName + '_error' );
 	const successEl = block.querySelector( '.' + blockClassName + '_success' );
+	emailField.addEventListener(
+		'input',
+		debounce( () => {
+			validateEmail( form, emailField );
+		}, 1000 )
+	);
 	form.addEventListener( 'submit', e => {
 		e.preventDefault();
-		const emailField = form.querySelector( 'input[name=email]' );
+		const email = emailField.value;
```

Comment:
We're declaring `const email` in validateEmail. We don't need it here, right?
```suggestion

```

Reply:
This is used by the 
```
emailField.addEventListener(
		'input',
		debounce( () => {
			validateEmail( form, emailField );
		}, 1000 )
	);
```

it is an enhancement so that errors are reported as you type (and when you click submit). We can remove it if you think we should.

Reply:
Oh right, I didn't catch that instance further down. We'll definitely want to leave that in there then! 👍